File: entity.js
var Entity = function() {
  this.x = 0;
  this.speed = 2; // units/s
  this.position_buffer = [];
}

// Apply user's input to this entity.
Entity.prototype.applyInput = function(input) {
  this.x += input.press_time*this.speed;
}

File: entity.go
package netgame

// Entity represents a game object
type Entity struct {
	ID int
	X float64
	Speed float64
	PositionBuffer []PositionState
}

// PositionState records where an entity was at a given time
type PositionState struct {
	time float64
	position float64
}

// ApplyInput is used by the server and by client interpolation to move the entity
func (e *Entity) ApplyInput(input Input) {
	e.X += float64(input.PressTime) * 0.1
}
